### ISSUE TITLE
Bug: cannot print a Configuration with a Constant that is a list or a tuple

### ISSUE THREAD

**Issue by mdorier** [automl/ConfigSpace#381]
Example:

```
from ConfigSpace import ConfigurationSpace, Constant, Integer

cs = ConfigurationSpace()
tags = ["a", "b", "c"]
cs.add(Constant('tags', tags))
config = cs.sample_configuration()
print(config)
```

The `print` statement throws this:

```
Traceback (most recent call last):
  File "/home/ubuntu/Code/mochi-bedrock/build/ttt.py", line 10, in <module>
    print(cs.sample_configuration())
  File "/home/ubuntu/Code/spack/var/spack/environments/bedrock-env/.spack-env/view/lib/python3.11/site-packages/ConfigSpace/configuration.py", line 247, in __str__
    values = dict(self)
             ^^^^^^^^^^
  File "/home/ubuntu/Code/spack/var/spack/environments/bedrock-env/.spack-env/view/lib/python3.11/site-packages/ConfigSpace/configuration.py", line 226, in __getitem__
    value = hyperparameter.to_value(vector)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/Code/spack/var/spack/environments/bedrock-env/.spack-env/view/lib/python3.11/site-packages/ConfigSpace/hyperparameters/hyperparameter.py", line 357, in to_value
    value: DType = self._transformer.to_value(np.array([vector]))[0]
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/Code/spack/var/spack/environments/bedrock-env/.spack-env/view/lib/python3.11/site-packages/ConfigSpace/hyperparameters/hp_components.py", line 516, in to_value
    return np.full_like(vector, self.value, dtype=object)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/Code/spack/var/spack/environments/bedrock-env/.spack-env/view/lib/python3.11/site-packages/numpy/core/numeric.py", line 407, in full_like
    multiarray.copyto(res, fill_value, casting='unsafe')
ValueError: could not broadcast input array from shape (3,) into shape (1,)
```

**Comment by eddiebergman** [comment]
Should be fixed in #382, seems the same dummy code for categoricals and ordinals with nested lists also triggered some bad behaviour. This is fixed now by sticking to lists for nested objects...

**Comment by eddiebergman** [comment]
Released in `1.1.3`
